tests/cli_tests.rs

```diff
@@ -45,6 +46,34 @@ fn make_tests() -> Result<()> {
         Test::basic("tasks"),
         Test::witharg("tasks-slvr", "tasks", "-slvr"),
         Test::basic("log"),
+        // don't need to rerun all of the cmds, just try to exercise each component of the core,
+        // registers, memory, start, stop
+        // also run each qemu command with the a different port so they can be parallelized
+        Test::witharg(
+            "qemu-readvar",
+            "qemu",
+            // these strings are weird since we have to escape the string so it is correctly
+            // written to the toml, but the toml itself also needs the escape characters for when
+            // it is parsed.
+            "--port 3333 --command \\\"readvar CLOCK_FREQ_KHZ\\\"",
+        ),
+        Test::witharg(
+            "qemu-halt",
+            "qemu",
+            " --port 3334 --command \\\"halt\\\"",
+        ),
+        Test::witharg(
+            "qemu-resume",
+            "qemu",
+            "--port 3335 --command \\\"resume\\\"",
+        ),
+        Test::witharg(
+            "qemu-dump",
+            "qemu",
+            "--port 3336 --command \\\"dump\\\"",
+        ),
+        //TODO how to verify halt/resume actually work
```

Comment:
jira issues https://rivosinc.atlassian.net/browse/SW-1169 and https://rivosinc.atlassian.net/browse/SW-1170 track these TODOs